backend/src/apportionment/fraction.rs

```diff
@@ -0,0 +1,159 @@
+use crate::data_entry::Count;
+use std::fmt;
+use std::fmt::{Debug, Display, Formatter};
+
+pub struct Fraction {
+    numerator: u64,
+    denominator: u64,
+}
+
+impl Fraction {
+    pub fn new(numerator: u64, denominator: u64) -> Self {
+        Self {
+            numerator,
+            denominator,
+        }
+    }
+
+    pub fn from_count(numerator: Count) -> Self {
+        Self::new(numerator as u64, 1)
+    }
+
+    pub fn from_u64(numerator: u64) -> Self {
+        Self::new(numerator, 1)
+    }
+
+    pub fn divide(&self, other: &Self) -> Self {
+        Self {
+            numerator: self.numerator * other.denominator,
+            denominator: self.denominator * other.numerator,
+        }
+    }
+
+    // divide by whole number and return the whole number
+    pub fn divide_whole(&self, other: &Self) -> u64 {
+        (self.numerator * other.denominator) / (self.denominator * other.numerator)
+    }
+}
+
+impl PartialOrd for Fraction {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Fraction {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        // compare using integer math
+        let lhs = self.numerator * other.denominator;
+        let rhs = self.denominator * other.numerator;
+        lhs.cmp(&rhs)
+    }
+}
+
+impl PartialEq for Fraction {
+    fn eq(&self, other: &Self) -> bool {
+        self.numerator * other.denominator == self.denominator * other.numerator
+    }
+}
+
+impl Eq for Fraction {}
+
+impl Display for Fraction {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        if self.denominator == 0 {
+            return write!(f, "NaN");
+        }
+        let whole_part = self.numerator / self.denominator;
+        let remainder = self.numerator % self.denominator;
+        if whole_part > 0 {
+            if remainder > 0 {
+                write!(f, "{} {}/{}", whole_part, remainder, self.denominator)
+            } else {
+                write!(f, "{}", whole_part)
+            }
+        } else {
+            write!(f, "{}/{}", self.numerator, self.denominator)
+        }
+    }
+}
+
+impl Debug for Fraction {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::apportionment::fraction::Fraction;
+
+    #[test]
+    fn test_from_count() {
+        let numerator = 5 as Count;
+        let fraction = Fraction::new(5, 1);
+        assert_eq!(fraction, Fraction::from_count(numerator));
+        assert_eq!(fraction.to_string(), "5")
+    }
```

Comment:
The parameters to `assert_eq!` in these tests is not consistent, mixing up the values passed to the left and right. I'd put the to-be-tested value on the left and the expected value on the right, e.g.:

```suggestion
    #[test]
    fn test_from_count() {
        let count = 5 as Count;
        let fraction = Fraction::from_count(count));
        assert_eq!(fraction, Fraction::new(5, 1));
        assert_eq!(fraction.to_string(), "5")
    }
```

And same for the other tests.